mathics/builtin/string/regexp.py

```diff
@@ -24,9 +24,9 @@ class RegularExpression(Builtin):
      = RegularExpression[[abc]]
 
     ## Mathematica doesn't seem to verify the correctness of regex
-    #> StringSplit["ab23c", RegularExpression["[0-9]++"]]
-     : Element RegularExpression[[0-9]++] is not a valid string or pattern element in RegularExpression[[0-9]++].
-     = StringSplit[ab23c, RegularExpression[[0-9]++]]
+    #> StringSplit["ab23c", RegularExpression["[0-9]++)"]]
+     : Element RegularExpression[[0-9]++)] is not a valid string or pattern element in RegularExpression[[0-9]++)].
```

Comment:
Still, maybe the best would be to move this to a pytest.

Reply:
I totally agree. The long-term plan is to remove all of the `#>` tests and move these to pytests. 

And the other aspect here, is that we should be testing `to_regex` more thoroughly there as well. 

When I say that the code is monolithic, part of what this means is that we have this module which contains the class definition for `StringSplit` as well as has regex conversion routine (that should be in `mathics.core.convert.regex`) and we test this as part of StringSplit instead of testing `to_regex()` directly.  

Testing via StringSplit makes it harder to pinpoint to_regex problems because we are going through layers of code  in order to get to the parts that we want (and want to strengthen) and when you find these the component you want to strengthen it is mingled with lots of other layers that are of different concerns. 

And then we also have a great deal code like this. This is just one of _several_ places where we have this. 

Other unrelated thing, there are many callers of `to_regex()` that are not quite right because they do not test to see if the result is None.  Those callers should be fixed.